superset/common/query_object.py

```diff
@@ -271,6 +288,10 @@ def cache_key(self, **extra: Any) -> str:
         """
         cache_dict = self.to_dict()
         cache_dict.update(extra)
+        if self.datasource:
+            cache_dict["datasource"] = self.datasource.uid
+        if self.result_type:
+            cache_dict["result_type"] = self.result_type
```

Comment:
These are only added to the dict if defined to avoid invalidating cache keys of currently cached objects.